### PR TITLE
Remove CircleCI from CI workflows

### LINKED ISSUE #947: Use github actions for windows CI tests
**Is your feature request related to a problem? Please describe.**

CircleCI has a limit on the number of hours we are allowed to use, when we run out of hours the windows python tests don't run anymore.

**Describe the solution you'd like**

Port the windows tests that currently run on CircleCI to github actions with the unit tests that are run on other platforms.

**Describe alternatives you've considered**

Only running windows tests after all other checks pass is another option here to reduce our usage of time on CircleCI.

### PR DESCRIPTION
**Pull request by tommy-waltmann**
## Description

This PR will remove CircleCI from all CI workflows, converting the windows tests to use github actions. I would also like this PR to reformat the current github actions CI to be more modular, and either fix or remove code coverage.

## Motivation and Context

CircleCI workflows have been blocked because there are no credits on our plan. Github Actions is a better CI deal anyway, so its time to convert freud's CI workflows.

Coverage has been broken since our move to scikit-build.

Resolves: #947 #699 

There is also an open pull request #749 which attempts to make the current CI format work with windows. I think this PR should make more sweeping CI changes than that PR would provide, so once this PR gets merged, that branch can be removed.

## How Has This Been Tested?

We will push commits and make sure the CI acts as we want it to. To test CI related to releases, we will need to do some more investigation of the build artifacts. I am honestly not sure how to test that the release process works without making a release.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds or improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation improvement (updates to user guides, docstrings, or developer docs)

## Checklist:

- [ ] I have read the [**CONTRIBUTING**](https://github.com/glotzerlab/freud/blob/master/CONTRIBUTING.md) document.
- [ ] My code follows the code style of this project.
- [ ] I have updated the documentation (if relevant).
- [ ] I have added tests that cover my changes (if relevant).
- [ ] All new and existing tests passed.
- [ ] I have updated the [credits](https://github.com/glotzerlab/freud/blob/master/doc/source/reference/credits.rst).
- [ ] I have updated the [Changelog](https://github.com/glotzerlab/freud/blob/master/ChangeLog.md).

### CODE REVIEW

File: old-github-actions/build_wheels.old [hunk 1]
```diff
@@ -2,7 +2,9 @@ name: Build wheels
 
 on:
   # Trigger on pull requests.
-  pull_request:
+  # commenting this out now so we can focus on new tests
+  # TODO remove this file before merging
+  #pull_request:
```
Comment (joaander) [inline old-github-actions/build_wheels.old]: You don't need to uncomment this, you could remove it if you are concerned about the duplicate work or extra time to use the new `run_tests` and cibuildwheel. This way, cibuildwheel to only test on `master` after PRs are merged.

You need to keep cibuildwheel as it builds and uploads the wheels to pypi.

File: .github/workflows/run_tests.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+on:
+  # trigger on pull requests
+  pull_request:
+
+  # trigger on all commits to master and tagged commits
+  push:
+    branches:
+      - "master"
+    tags:
+      - "v*"
```
Comment (joaander) [inline .github/workflows/run_tests.yml]:
```suggestion
```
If you aren't uploading wheels in this workflow, there is no need to run it on tags.

**Review by joaander — COMMENTED** [review]
I know this is a draft, but I thought I'd offer a few suggestions. The new multi-platform `run_tests.yml` is really clean!

File: old-github-actions/build_wheels.old (hunk 1 above)
Comment (tommy-waltmann) [reply old-github-actions/build_wheels.old]: I'll move it back to where it was. I will also uncomment for now so I can make sure it works with the new format, but I don't want it to trigger on PRs when this is PR is eventually merged.

**Comment by tommy-waltmann** [comment]
We have a failing test right now which fails because on #950 the command `target_link_options` was used in the `make_python_extension_module` function. That command was added in cmake 3.13.0, and our oldest ci requirement for cmake is currently 3.12.0.

This should have triggered an error in old way of testing which was to just build and test the wheels. In that test, cmake 3.22.0 appears to be installed and used when testing against python 3.6 instead of our oldest required cmake.

I am making a note of this now in order to further investigate later.

EDIT: Upon further inspection, the reason this happens is because we install a version of cmake with no version requirement via pip in order to build tbb from source in the wheel.

**Comment by joaander** [comment]
If you want to maintain support for 3.13, you can add a condition on the cmake version and use the pre-3.13 command for link options.

**Comment by tommy-waltmann** [comment]
to my knowledge, maintaining support for 3.12 isn't especially important so I'm fine with bumping the oldest requirement.

**Comment by joaander** [comment]
@tommy-waltmann 

> I am honestly not sure how to test that the release process works without making a release.

You have the `workflow_dispatch:` trigger. You can manually trigger the `build_wheels.yml` workflow on a branch before merging and tagging: https://github.blog/changelog/2020-07-06-github-actions-manual-triggers-with-workflow_dispatch/

This is how I tested #962 before opening the pull request.

**Comment by tommy-waltmann** [comment]
Things to do before this PR is ready for review:

- [x] Remove the duplicate requirements files which are now hanging around, and decide where the files that are kept should be located
- [ ] Discuss adding the benchmarks back in as tests, only run them upon adding a benchmark label to the PR (similar to how hoomd uses the validate label)
- [x] Either figure out how to remove circleci from trying to run tests, or create a dummy circleci config file which always passes

**Comment by tommy-waltmann** [comment]
After a bit of thought, I would like this PR to get merged without adding the scripts needed to run the benchmarks. It appears those will take a significant amount of more work to get that in place, and I'd like to have the new unit testing scripts in place soon.

Adding the benchmarks will be deferred to a different PR, and I've made an issue to track that this needs to be done in #976.

File: .github/workflows/conda-envs/requirements-test-oldest.txt [hunk 3]
```diff
@@ -1,5 +1,5 @@
 ase==3.16.0
```
Comment (bdice) [inline .github/workflows/conda-envs/requirements-test-oldest.txt]: We need to name this file something besides `requirements` -- Dependabot triggers version updates on all files named `requirements*.txt`. We don't want Dependabot to update our oldest dependencies. You could name it `test-oldest.txt` or `oldest-reqs.txt` or something like that.

(I can't find the reference that lists the pattern anymore -- I believe I had to search Dependabot's source code the first time I set this up. I also can't find a reference to this in the PRs where I added this file, but I thought I had left a comment somewhere for historical reasons... because it did take some time to track down.)

File: .github/workflows/build_wheels.yml [hunk 4]
```diff
@@ -1,9 +1,6 @@
 name: Build wheels
 
 on:
-  # Trigger on pull requests.
-  pull_request:
```
Comment (bdice) [inline .github/workflows/build_wheels.yml:5]: Can we continue to use cibuildwheel to build/test macOS and Linux, and only use the `run_tests.yml` script for Windows? I strongly prefer testing with cibuildwheel wherever possible because it ensures consistency in our building/testing process between releases and normal commits. That reduces the risk of build problems emerging at release time.

To do that, we'd need to re-enable `pull_request` triggers for `build_wheels.yml` and disable `macOS-latest, ubuntu-latest` runners on `run_tests.yml`.

File: .github/workflows/build_wheels.yml [hunk 5]
```diff
@@ -79,7 +76,7 @@ jobs:
           python-version: '3.9'
 
       - name: Install build requirements
-        run: python3 -m pip --disable-pip-version-check install scikit-build cmake -r requirements.txt --upgrade --progress-bar=off
+        run: python3 -m pip --disable-pip-version-check install scikit-build cmake -r .github/workflows/conda-envs/requirements-test.txt --upgrade --progress-bar=off
```
Comment (bdice) [inline .github/workflows/build_wheels.yml]: The requirements for _building_ should be minimal. The full set of testing requirements should not be installed at build time.

File: .github/workflows/cibuildwheel-before-test.sh [hunk 6]
```diff
@@ -8,12 +8,12 @@ PACKAGE_DIR=$1
 
 if [[ $(python --version 2>&1) == *"3.6."* ]]; then
   # Python 3.6 is only supported with oldest requirements
-  pip install -U -r "${PACKAGE_DIR}/.circleci/ci-oldest-reqs.txt" --progress-bar=off
+  pip install -U -r "${PACKAGE_DIR}/.github/workflows/conda-envs/requirements-test-oldest.txt" --progress-bar=off
 elif [[ $(python --version 2>&1) == *"3.7."* ]]; then
   # Python 3.7 was dropped by NEP 29 so not compatible with some of the newest dependencies
-  pip install -U -r "${PACKAGE_DIR}/requirements/requirements-test-compatible.txt" --progress-bar=off
+  pip install -U -r "${PACKAGE_DIR}/.github/workflows/conda-envs/requirements-test-compatible.txt" --progress-bar=off
```
Comment (bdice) [inline .github/workflows/cibuildwheel-before-test.sh]: I would recommend dropping Python 3.7 rather than continuing to maintain a separate set of "compatible" dependencies.

File: .github/workflows/conda-envs/test-oldest_env.yaml [hunk 7]
```diff
@@ -0,0 +1,12 @@
+name: test
+channels:
+  - conda-forge
+  - defaults
+
+dependencies:
+  - tbb
```
Comment (bdice) [inline .github/workflows/conda-envs/test-oldest_env.yaml]: `tbb`, `tbb-devel` should be pinned to the minimum supported version here. [The docs claim](https://freud.readthedocs.io/en/latest/gettingstarted/installation.html) compatibility with `>=2017.03.R2` but I am unsure if that is still accurate since the TBB 2020 updates.

(I don't really care about pinning ninja/pip since they're not build dependencies. One could make the argument that pip matters because various behaviors have changed over time, but I don't feel strongly about pinning it.)

File: .github/workflows/run_tests.yml [hunk 8]
```diff
@@ -0,0 +1,58 @@
+name: Run Unit Tests
+
+# Remove this for now, circleci is getting in the way with tests failing
```
Comment (bdice) [inline .github/workflows/run_tests.yml]: Is this comment still relevant? Is it saying that a failure from CircleCI caused the GitHub Actions builds to fail? Seems like this may be resolved but I'm unsure.

File: .github/workflows/run_tests.yml [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  # trigger on pull requests
+  pull_request:
+
+  # trigger on all commits to master and tagged commits
```
Comment (bdice) [inline .github/workflows/run_tests.yml]: I'd like to see the project rename `master` ->` main` in alignment with a large portion of the Python ecosystem and open source community. Maybe this is as good a time as any.

File: .github/workflows/run_tests.yml [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+  push:
+    branches:
+      - "master"
+
+  # trigger on request
+  workflow_dispatch:
+
+jobs:
+  run_tests:
+    name: Run tests on ${{ matrix.os }} with python ${{ matrix.pyver-env.pyver }}
```
Comment (bdice) [inline .github/workflows/run_tests.yml]:
```suggestion
    name: Run tests on ${{ matrix.os }} with Python ${{ matrix.pyver-env.pyver }}
```

File: .github/workflows/run_tests.yml [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [macOS-latest, ubuntu-latest, windows-latest]
+        pyver-env: [ {pyver: '3.6', env: 'test-oldest_env.yaml'},
+                     {pyver: '3.7', env: 'test-compatible_env.yaml'},
+                     {pyver: '3.8', env: 'test_env.yaml'},
+                     {pyver: '3.9', env: 'test_env.yaml'},
+                     {pyver: '3.10', env: 'test_env.yaml'} ]
```
Comment (bdice) [inline .github/workflows/run_tests.yml:30]: 3x5 jobs is a pretty large support matrix. We can shrink this to 3 operating systems by 4 configurations (3.8 oldest, 3.8, 3.9, 3.10).

File: .github/workflows/run_tests.yml [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+        uses: actions/checkout@v2.3.4
+        with:
+          submodules: true
+
+      - name: Create Python Environment
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          python-version: ${{ matrix.pyver-env.pyver }}
+          environment-file: .github/workflows/conda-envs/${{ matrix.pyver-env.env }}
+          channels: conda-forge,defaults
```
Comment (bdice) [inline .github/workflows/run_tests.yml]: I would remove `defaults` from all conda channels. We should pull exclusively from `conda-forge` to avoid incompatibilities.

Comment (bdice) [reply .github/workflows/run_tests.yml]: I'd recommend using mambaforge here: https://github.com/marketplace/actions/setup-miniconda#example-10-miniforge

File: .github/workflows/style_check.yml [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+
+  # Trigger on request.
+  workflow_dispatch:
+
+jobs:
+  clang-tidy:
+    name: Check C++ Code Style
+    runs-on: ubuntu-latest
+    container:
+      image: glotzerlab/ci:2021.11-clang10_py38
```
Comment (bdice) [inline .github/workflows/style_check.yml:24]: We could bump this to [2022.05-clang10_py38](https://hub.docker.com/layers/ci/glotzerlab/ci/2022.05-clang10_py38/images/sha256-ec335538528cdd482f74d531ea1503225fa337267965849d771371c9b9453a9c?context=explore) while we're editing this.
```suggestion
      image: glotzerlab/ci:2022.05-clang10_py38
```

File: .github/workflows/style_check.yml [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+    steps:
+      - uses: actions/checkout@v2.3.4
```
Comment (bdice) [inline .github/workflows/style_check.yml]:
```suggestion
      - uses: actions/checkout@v3.0.2
```

File: .github/workflows/style_check.yml [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+        with:
+          submodules: true
+      - name: Install pre-commit
+        run: python3 -m pip install pre-commit
+      - name: Build freud and export compile commands
+        run: |
+          python3 -m pip install scikit-build
```
Comment (bdice) [inline .github/workflows/style_check.yml:35]: Is this only installing scikit-build because it's not in the CI image but the other dependencies (Cython, TBB) are already available?

**Review by bdice — COMMENTED** [review]
Great work! CI is very hard to deal with and I appreciate your efforts on this. I have a few comments for improvement.

File: .github/workflows/build_wheels.yml (hunk 4 above)
Comment (tommy-waltmann) [reply .github/workflows/build_wheels.yml:5]: That's a fair concern. The reason I originally only wanted to use `run_tests.yml` for regular commits was so we don't have to build tbb from source just to unit test. The tradeoff is CI time vs the release process working security.

I don't believe hoomd runs cibuildwheel on every commit, so maybe @joaander can weigh in on this idea before we decide how to move forward.

File: .github/workflows/build_wheels.yml (hunk 5 above)
Comment (tommy-waltmann) [reply .github/workflows/build_wheels.yml]: Would you rather I install just the build requirements to build, and then before testing do another `pip install` for the test-specific requirements? We need the testing requirements to be installed at some point.

File: .github/workflows/cibuildwheel-before-test.sh (hunk 6 above)
Comment (tommy-waltmann) [reply .github/workflows/cibuildwheel-before-test.sh]: Good point. My plan was to re-evaluate which python versions we support once python 3.11 was released. Dropping support for python versions might be best for another PR, though.

File: .github/workflows/style_check.yml (hunk 15 above)
Comment (tommy-waltmann) [reply .github/workflows/style_check.yml:35]: Yes.

File: .github/workflows/run_tests.yml (hunk 8 above)
Comment (tommy-waltmann) [reply .github/workflows/run_tests.yml]: It's not needed anymore, I'll remove this.

File: .github/workflows/build_wheels.yml (hunk 4 above)
Comment (joaander) [reply .github/workflows/build_wheels.yml:5]: I prefer the simplicity of the single action that tests on all platforms. This is easier to update and keep consistent. As Tommy mentioned, it also reduces the CI times run on every commit and tests that users can build and install freud against conda-forge prereqs. `cibuidwheel` tests a very narrow environment matching a specific set of old OSes and specific dependencies provided by `pip`.

I'm in favor of deferring the `cibuildwheel` tests to limited runs at release time, or run once just before merging a PR. In `gsd` I do run the `cibuildwheel` builds AND the action based builds on every commit, but gsd is very quick to build and test.

File: .github/workflows/build_wheels.yml (hunk 5 above)
Comment (bdice) [reply .github/workflows/build_wheels.yml]: Yes, that's exactly what I would recommend.

File: .github/workflows/style_check.yml (hunk 15 above)
Comment (bdice) [reply .github/workflows/style_check.yml:35]: A comment indicating that would be beneficial.

File: .github/workflows/run_tests.yml (hunk 9 above)
Comment (tommy-waltmann) [reply .github/workflows/run_tests.yml]: I've never understood to motivation for such a change, personally. If that change would make the package more professional looking, though, I can support it. For the sake of PR modularity, it may be best suited for another PR, but I'm still on the fence about renaming it.

File: .github/workflows/run_tests.yml (hunk 11 above)
Comment (tommy-waltmann) [reply .github/workflows/run_tests.yml:30]: See an earlier comment for my opinion on python version support.

File: .github/workflows/build_wheels.yml (hunk 4 above)
Comment (bdice) [reply .github/workflows/build_wheels.yml:5]: I haven't looked at the CI run times but that seems like a good argument for not using cibuildwheel if it's substantially different. I agree that being able to test against conda-forge dependencies is helpful. My preference for consistent CI environments between normal commits and releases is partly driven by the headaches I faced in making releases of the package in the past. Things are much more stable and better tested now, so do whatever you think is best here. I'd say running cibuildwheel on PR merge (every commit to the main branch) is a good compromise. The CI build time on the main branch is usually not a blocker, and it ensures that tests are performed regularly and problems can be caught before release.

File: .github/workflows/run_tests.yml (hunk 9 above)
Comment (bdice) [reply .github/workflows/run_tests.yml]: These resources help explain why projects are making this move. I'm happy with keeping PRs modular, it doesn't have to be done now but should be considered in the near future. signac plans to do the same migration, probably around the time of the 2.0 release.
https://github.com/github/renaming
https://www.theserverside.com/feature/Why-GitHub-renamed-its-master-branch-to-main
https://www.zdnet.com/article/github-to-replace-master-with-main-starting-next-month/

File: .github/workflows/build_wheels.yml (hunk 4 above)
Comment (joaander) [reply .github/workflows/build_wheels.yml:5]: Tommy can comment on the time needed to run the cibuildwheel tests. There are cases to testing both cibuildwheel release tests and builds on modern operating systems with current (and oldest) conda-forge dependencies.

CI hours aren't a problem, GitHub provides us with 60 concurrent runners (organization wide) - though only 5 mac runners. For me, it is a question of what is the lag time from push to CI test approval. Make that too long and it hampers productivity.

Comment (tommy-waltmann) [reply .github/workflows/build_wheels.yml:5]: Just looked at the CI times. Comparing the latest commit on this branch to the latest commit on the master branch: linux tests seem to take ~5m on this branch vs. ~7m on master. Time for Mac tests vary significantly (because of runner limits) but range between ~9-13m on this branch and ~10-17m on master.

It appears there is some slight time benefit to not building TBB from source in the new tests.

My suggestion going forward would be to have all commits on master build wheels, and then using the `run_tests.yml` script only on regular commits.

Comment (bdice) [reply .github/workflows/build_wheels.yml:5]: Awesome. Sounds good!

**Comment by tommy-waltmann** [comment]
Bumping the ci image from 2021.11 to 2022.05 seems to cause issues when trying to lint the C++ code. Will investigate further on this later.

File: .github/workflows/run_tests.yml (hunk 12 above)
Comment (tommy-waltmann) [reply .github/workflows/run_tests.yml]: What's the benefit of using `Mambaforge` here instead of just setting channels to conda-forge exclusively?

File: .github/workflows/conda-envs/test-oldest_env.yaml (hunk 7 above)
Comment (tommy-waltmann) [reply .github/workflows/conda-envs/test-oldest_env.yaml]: I'll reverse engineer what our oldest supported is on this PR.

Comment (tommy-waltmann) [reply .github/workflows/conda-envs/test-oldest_env.yaml]: Looking through tbb's versions on conda-forge, I don't see a version matching that name. In fact, the development versions only go back as far as `2018.0.4`. I'll try to pin to that version for now and see what happens. We'll have to update the oldest supported version in our docs.

File: .github/workflows/run_tests.yml (hunk 12 above)
Comment (bdice) [reply .github/workflows/run_tests.yml]: Using mambaforge means that you will be getting mamba by default. `mamba` solves environments and installs dramatically faster than `conda`.

**Comment by tommy-waltmann** [comment]
2019.7 seems to be our oldest supported version of tbb.

File: .circleci/config.yml [hunk 16]
```diff
@@ -1,251 +1,7 @@
...
-      - windows-python-37:
-          requires:
-            - check-style
-            - windows-python-310
-# Disabled benchmarks on CI - they currently take too much work
-#     - benchmarks:
-#         requires:
-#           - linux-python-38
+    steps:
+      - run: echo "Nothing to do."
```
Comment (bdice) [inline .circleci/config.yml]: Just delete this file if it doesn't do anything.

File: .github/workflows/cibuildwheel-before-test.sh [hunk 17]
```diff
@@ -8,12 +8,12 @@ PACKAGE_DIR=$1
 
 if [[ $(python --version 2>&1) == *"3.6."* ]]; then
   # Python 3.6 is only supported with oldest requirements
-  pip install -U -r "${PACKAGE_DIR}/.circleci/ci-oldest-reqs.txt" --progress-bar=off
+  pip install -U -r "${PACKAGE_DIR}/.github/workflows/conda-envs/oldest-test-requirements.txt" --progress-bar=off
```
Comment (bdice) [inline .github/workflows/cibuildwheel-before-test.sh]: I'm not sure, but this may still trigger Dependabot. I named the old file `reqs` instead of `requirements`.

File: .github/workflows/conda-envs/oldest-test-requirements.txt [hunk 18]
```diff
@@ -1,5 +1,5 @@
 ase==3.16.0
-cmake==3.12.0
+cmake==3.13.0
```
Comment (bdice) [inline .github/workflows/conda-envs/oldest-test-requirements.txt]: Why was this bumped?

File: .github/workflows/conda-envs/test-compatible_env.yaml [hunk 19]
```diff
@@ -0,0 +1,12 @@
+name: test
+channels:
+  - conda-forge
+  - defaults
```
Comment (bdice) [inline .github/workflows/conda-envs/test-compatible_env.yaml]: I would remove `defaults` from all these environments.

File: .github/workflows/run_tests.yml [hunk 20]
```diff
@@ -0,0 +1,57 @@
...
+  cancel-in-progress: true
+
+on:
+  # trigger on pull requests
+  pull_request:
+
+  # trigger on all commits to master and tagged commits
+  push:
+    branches:
+      - "master"
```
Comment (bdice) [inline .github/workflows/run_tests.yml:14]: The comment says this runs on all tagged commits, but this doesn't do that.

**Review by bdice — APPROVED** [review]
A few suggestions -- great work.

File: .circleci/config.yml (hunk 16 above)
Comment (tommy-waltmann) [reply .circleci/config.yml]: For some reason, circleci still tries to run even if I remove the config file and it shows up on github as a failed test. The dummy makes sure we can still get the coveted green check mark.

File: .github/workflows/conda-envs/oldest-test-requirements.txt (hunk 18 above)
Comment (tommy-waltmann) [reply .github/workflows/conda-envs/oldest-test-requirements.txt]: This was needed because of a change made in #950 (calling `target_link_options`). I don't remember why it didn't get caught in that PR, but I'm fixing it here.

File: .circleci/config.yml (hunk 16 above)
Comment (bdice) [reply .circleci/config.yml]: Ah. That just takes some fixing on the GitHub / CircleCI website side of things. Go ahead and delete it, and let me know if you want help fixing that.

File: .github/workflows/build_wheels.yml [hunk 21]
```diff
@@ -1,9 +1,6 @@
 name: Build wheels
 
 on:
-  # Trigger on pull requests.
-  pull_request:
-
```
Comment (joaander) [inline .github/workflows/build_wheels.yml:6]: This comment applies to line 126.

Now that you are running these tests only on the trunk branches, consider adjusting the slack notification to show you when wheel builds fail. The alternative is to regularly check the GitHub interface or README badge to see if tests are passing on the trunk branches.

You can replace `if: startsWith(github.ref, 'refs/tags/v') && always()` with `if: startsWith(github.ref, 'refs/tags/v') || failure()` to get notified when tests fail.

File: .github/workflows/conda-envs/oldest-test-reqs.txt [hunk 22]
```diff
@@ -1,5 +1,5 @@
 ase==3.16.0
-cmake==3.12.0
```
Comment (joaander) [inline .github/workflows/conda-envs/oldest-test-reqs.txt]: I wouldn't put `pip` requirements files in a directory named `conda-envs`. Developers would expect to find conda environment files in such a directory.

File: .github/workflows/conda-envs/test-compatible_env.yaml [hunk 23]
```diff
@@ -0,0 +1,11 @@
...
+channels:
+  - conda-forge
+
+dependencies:
+  - tbb
+  - tbb-devel
+  - ninja
+  - pip
+  - pip:
+    - -r requirements-test-compatible.txt
```
Comment (joaander) [inline .github/workflows/conda-envs/test-compatible_env.yaml]: I'm surprised this works. conda-forge provides packages with a specific ABI that is not the same as PyPI's manylinux. Prefer the conda-forge provided packages and only fall back on pip installations when absolutely necessary. In such cases, consider contributing the package to the conda-forge ecosystem.

File: MANIFEST.in [hunk 24]
```diff
@@ -2,7 +2,6 @@ recursive-include cpp *.cc *.h CMakeLists.txt
 recursive-include CMake *
 recursive-include extern *
 recursive-include freud *.py *.pyx *.pxd *.pxi CMakeLists.txt
-recursive-include requirements
```
Comment (joaander) [inline MANIFEST.in:5]: Why were these included before? If there was a good reason, why not package them now?

**Review by joaander — APPROVED** [review]
Thanks, this is a great improvement.

Since the new tests are passing and circleci has become troublesome I don't want to delay this merge. I do offer a few optional suggestions for improvement.

File: MANIFEST.in (hunk 24 above)
Comment (bdice) [reply MANIFEST.in:5]: Including those files was an arbitrary decision I made. I don’t think there is a compelling reason to include requirements files.

File: .github/workflows/conda-envs/oldest-test-reqs.txt (hunk 22 above)
Comment (tommy-waltmann) [reply .github/workflows/conda-envs/oldest-test-reqs.txt]: I could rename the directory `env-files`, or something like that.

File: .github/workflows/conda-envs/test-compatible_env.yaml (hunk 23 above)
Comment (tommy-waltmann) [reply .github/workflows/conda-envs/test-compatible_env.yaml]: `run_tests.yml` has to install the test dependencies plus tbb, but `build_wheels.yml` has to install the test dependencies without tbb. I could not think of any other way to have two separate sets of requirements files without writing the same pinned dependencies in two different places.
